### PR TITLE
LibGUI: Allow strings to be uppercased, lowercased and snakecased

### PR DESCRIPTION
**Pull request by qcerasmus**
Added three shortcuts on TextEditor:
1) ctrl+shift+U will uppercase selected text
2) ctrl+shift+L will lowercase selected text
3) ctrl+shift+N will snake case selected text

### CODE REVIEW

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 1]
```diff
@@ -1111,6 +1121,45 @@ void TextEditor::unindent_line()
...
 }
 
+void TextEditor::uppercase_selection()
+{
+    if (!has_selection())
+        return;
+    auto const selection_start = m_selection.start() > m_selection.end() ? m_selection.end() : m_selection.start();
+    auto const selection_end = m_selection.end() > m_selection.start() ? m_selection.end() : m_selection.start();
+
+    auto const range = TextRange(selection_start, selection_end);
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibGUI/TextEditor.cpp:1131]: We already have the current selection handy:
```suggestion
    auto const range = selection();
```

(Same in `lowercase_selection()` and `snakecase_selection()`.)

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Hi there!

I noticed a couple of bugs while trying this out in TextEditor:
- Ctrl-Shift-U doesn't do anything.
- These case changes aren't added to the Undo stack correctly. I can't tell what's wrong, but I'm not familiar with the undo stack. Ask on Discord if you need help.

In GMLPlayground, none of the three shortcuts change the selection's case.

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 2]
```diff
@@ -1052,6 +1052,16 @@ void TextEditor::keydown_event(KeyEvent& event)
...
 
+    if (event.shift() && event.ctrl() && event.key() == KeyCode::Key_U) {
+        uppercase_selection();
+    }
+    if (event.shift() && event.ctrl() && event.key() == KeyCode::Key_L) {
+        lowercase_selection();
+    }
+    if (event.shift() && event.ctrl() && event.key() == KeyCode::Key_N) {
+        snakecase_selection();
+    }
```
[the remark this answers is not in the document]
Comment (qcerasmus) [reply Userland/Libraries/LibGUI/TextEditor.cpp:1063]: I've added this and will commit soon

File: Userland/Libraries/LibGUI/TextEditor.cpp (hunk 1 above)
Comment (qcerasmus) [reply Userland/Libraries/LibGUI/TextEditor.cpp]: The changes requested above have been done. For the snake_selection I did update the selected text as the snake case will be larger than the original text.

**Pull request closed without merging**